index.js

```diff
@@ -19,49 +20,62 @@ const SUBMISSION = [
 ]
 
 app.post('/signup', function(req, res) {
-  // Add logic to decode body
-  // body should have email and password
-
-
-  //Store email and password (as is for now) in the USERS array above (only if the user with the given email doesnt exist)
-
-
-  // return back 200 status code to the client
-  res.send('Hello World!')
+ 
+    const { email, password } = req.body;
+  const existingUser = USERS.find(user => user.email === email);
+  if (existingUser) {
+    return res.status(400).send('User already exists');
+  }
+  USERS.push({ email, password });
+  res.sendStatus(200);
 })
 
 app.post('/login', function(req, res) {
-  // Add logic to decode body
-  // body should have email and password
-
-  // Check if the user with the given email exists in the USERS array
-  // Also ensure that the password is the same
+     const { email, password } = req.body;
+    
+    const user = USERS.find(user => user.email === email);
+    if (!user) {
+      return res.status(401).send('Invalid email or password');
+    }
+   
+    if (user.password !== password) {
```

Comment:
`Optional 💡` when dealing with passwords standard practice is to hash the password and store in your data store. 

You can checkout [bcrypt](https://www.npmjs.com/package/bcrypt) if interested to implement with password hashing